src/view/geometry/Dimension.js

```diff
@@ -35,6 +35,16 @@ define([
      */
     function Dimension() {
         Base.apply(this, arguments);
+        var args = arguments;
+        if (args.length === 1 && args[0] instanceof Dimension) {
+            this.w = args[0].w;
+            this.h = args[0].h;
+        } else if (args.length === 2
+                && typeof args[0] === 'number'
+                && typeof args[1] === 'number') {
+            this.w = args[0];
+            this.h = args[1];
+        }
     }
 
     genetic.inherits(Dimension, Base, {
```

Comment:
I think w, h should have it's default value as 0
to prevent undefined w,h for constructor with blank parameters like the following.
````js
var dimension = new Dimension(); //without arguments
console.log(dimension.w); //undefined